### PR TITLE
Change Gen.double and Gen.float so that they take a Range type

### PR DESCRIPTION
**Pull request by moodmosaic**

### CODE REVIEW

**Comment by moodmosaic** [comment]
`Range` and `Shrink` are in line with the Haskell version, and with this PR `Seed` too. Next is `Gen` 😃

File: src/Hedgehog/Seed.fs [hunk 1]
```diff
@@ -144,6 +144,20 @@ module Seed =
...
+    let rec nextDouble (lo : double) (hi : double) (seed : Seed) : double * Seed =
+        if lo > hi then
+            nextDouble hi lo seed
+        else
+            let x, seed' =
+                seed |> nextBigInt
+                    (bigint Int32.MinValue)
+                    (bigint Int32.MaxValue)
+            let scaled_x =
+                  (0.5 * lo + 0.5 * hi) + ((0.5 * hi - 0.5 * lo) / (0.5 * 4294967296.0)) * float x
```
Comment (jystic) [inline src/Hedgehog/Seed.fs]: Curious where this algorithm came from / what is the thinking behind it?

Comment (moodmosaic) [inline src/Hedgehog/Seed.fs]: This is `randomIvalDouble` ported from Haskell [System.Random](https://hackage.haskell.org/package/random-1.1/docs/src/System-Random.html), similar to the existing `Seed.nextBigInt` which is really `randomIvalInteger` ported from that same module.

Comment (jystic) [inline src/Hedgehog/Seed.fs]: ah perfect 👍

**Comment by jystic** [comment]
✌️

File: src/Hedgehog/Seed.fs (hunk 1 above)
Comment (moodmosaic) [inline src/Hedgehog/Seed.fs]: :neckbeard: 😃

**Pull request merged**